### PR TITLE
A2

### PR DESCRIPTION
**Pull request by beccadaniel**

### CODE REVIEW

File: assignments/A2/Nim.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			
+			System.out.printf("\nA: %d\tB: %d\tC: %d\n", A, B, C);
+			
+			switch(A +B + C ){
+			case 0:
+				System.out.println("\n" + player2 + ", there are no counters left, so you WIN!");
+				continue;
+			}
+			do{
+	    		System.out.print("\n" + player2 + ", choose a pile: ");
```
Comment (lon-io) [inline assignments/A2/Nim.java:77]: Please move this logic to a separate method. It's repeated

File: assignments/A2/Nim.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			    	break;
+			    }
+			  }
+			
+			while(check==0);
+			
+			System.out.printf("How many to remove from pile %c: ", pile2);
+			remove = input.nextInt();
+			
+			switch(pile2){
```
Comment (lon-io) [inline assignments/A2/Nim.java:97]: This also is repeated, so a separate method is suitable

File: assignments/A2/Nim.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			case 'C':
+				C = C - remove;
+				if(C < 0)
+					C=0;
+				break;	
+			}
+			
+			System.out.printf("\nA: %d\tB: %d\tC: %d\n", A, B, C);
+			
+			switch(A +B + C ){
```
Comment (lon-io) [inline assignments/A2/Nim.java:71]: I reckon it would be better to assign A+B+C to a variable.

File: assignments/A2/Nim_Game.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+				break;
+			thePile = nimgame(player2, player1, thePile);
+
+			loopCondition = thePile[0]+thePile[1]+thePile[2];
+		}
+		while(loopCondition != 0);
+		
+	}	
+	
+	public static int[] nimgame(String currentPlayer, String nextPlayer, int[] thePile){
```
Comment (lon-io) [inline assignments/A2/Nim_Game.java:49]: This and other methods below will only be called within this class, so they can be `private`

File: assignments/A2/Nim_Game.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+		int remove, check=0;
+		char pile;
+		int conditionToEnd;
+		
+		Scanner input = new Scanner(System.in);
+		
+    	do{
```
Comment (lon-io) [inline assignments/A2/Nim_Game.java:56]: Please, fix your indentations, if only for Aesthetics :)

File: assignments/A2/Nim_Game.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
+import java.util.Scanner;
+
+public class Nim_Game{
+	
+	
+	public static void main(String[] args){
+		nim();
+	}
+	
+	public static void nim(){
```
Comment (lon-io) [inline assignments/A2/Nim_Game.java:10]: This can be `private` too

File: assignments/A2/Nim_Game.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
+import java.util.Scanner;
+
+public class Nim_Game{
+	
+	
+	public static void main(String[] args){
+		nim();
+	}
+
```
Comment (lon-albert) [inline assignments/A2/Nim_Game.java]: Please, add block comments before all your methods, makes it easier for you to remember what you were doing when you revisit this code anytime later. Plus, it saves me review time too

File: assignments/A2/Nim_Game.java (hunk 6 above)
Comment (lon-io) [inline assignments/A2/Nim_Game.java:10]: Please, add block comments before this method like you did for others.

File: assignments/A2/Nim_Game.java (hunk 4 above)
Comment (lon-io) [reply assignments/A2/Nim_Game.java:49]: Add block comments for this method too

**Review by lon-io — APPROVED** [review]

**Comment by lon-io** [comment]
Great job @beccadaniel !

**Pull request merged**